sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```
@@ -604,7 +604,13 @@ class Analyzer(
 
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
       case i @ InsertIntoTable(u: UnresolvedRelation, parts, child, _, _) if child.resolved =>
-        i.copy(table = EliminateSubqueryAliases(lookupTableFromCatalog(u)))
+        val newTable = EliminateSubqueryAliases(lookupTableFromCatalog(u))
+        // Inserting into a view is not allowed, we should throw an AnalysisException.
+        if (newTable.isInstanceOf[View]) {
```

The rule `ResolveRelations` executes before `PreprocessTableInsertion`, so we can fail early here.